### PR TITLE
Add Flink CLI: Add main.go and artifacts (commands: list, info, job, run, cancel)

### PR DESCRIPTION
**Pull request by OhRobin**
- upload is still not stable
- to test out the local build (running `dcos flink <commands>`

### CODE REVIEW

File: container/appmaster/cli/dcos-flink/main.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+type UploadHandler struct {
+	upload string
+}
+
+func (cmd *UploadHandler) runUpload(c *kingpin.ParseContext) error {
+	//TODO x509 auth instead of https to http change
+	url := client.OptionalCLIConfigValue("core.dcos_url") //TODO this should be a RequiredCLIConfigValue
+	url = strings.Replace(url,"https://", "http://", 1)
+	url = fmt.Sprintf("%s/service/flink/jars/upload", url)
```
Comment (joerg84) [inline container/appmaster/cli/dcos-flink/main.go]: flink should be variable name

File: container/appmaster/cli/dcos-flink/main.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+	return nil
+}
+
+
+func handleJobSection(app *kingpin.Application) {
+	cmd := &InfoHandler{}
+	job := app.Command("info", "Summary of Job status.").Action(cmd.runInfo)
+	job.Arg("jobid",
+					"Summary of one job").StringVar(&cmd.info)
```
Comment (joerg84) [inline container/appmaster/cli/dcos-flink/main.go]: weird indent

File: container/appmaster/cli/dcos-flink/main.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	"strings"
+	"net/http"
+	"io/ioutil"
+)
+
+
+func main() {
+	app := cli.New()
+
+	// cli.HandleDefaultSections(app)
```
Comment (joerg84) [inline container/appmaster/cli/dcos-flink/main.go]: remove

File: container/appmaster/cli/dcos-flink/main.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+	fmt.Println(string(body))
+	return nil
+}
+
+func handleUploadSection(app *kingpin.Application) {
+	cmd := &UploadHandler{}
+	upload := app.Command("upload", "Upload flink jar to run").Action(cmd.runUpload)
+	upload.Arg("jar file", "jar file to upload").Required().StringVar(&cmd.upload)
+}
```
Comment (joerg84) [inline container/appmaster/cli/dcos-flink/main.go]: Could you create a small section here about the CLI https://github.com/dcos/examples/blob/master/flink/1.9/README.md

File: container/appmaster/cli/dcos-flink/main.go [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+		client.PrintJSONBytes(response)
+	} else {
+		log.Println(err)
+	}
+	return nil
+}
+
+func handleCancelSection(app *kingpin.Application) {
+	cmd := &CancelHandler{}
+	cancel := app.Command("cancel", "cancel flink job").Action(cmd.runCancel)
```
Comment (joerg84) [inline container/appmaster/cli/dcos-flink/main.go]: You have inconsistent capitalization in the command description

**Comment by OhRobin** [comment]
Requesting Review with the following subcommands:
- list 
- info 
- job
- run
- cancel

**Comment by OhRobin** [comment]
@joerg84

File: container/appmaster/cli/dcos-flink/main.go (hunk 3 above)
Comment (OhRobin) [inline container/appmaster/cli/dcos-flink/main.go]: addressed

File: container/appmaster/cli/dcos-flink/main.go (hunk 2 above)
Comment (OhRobin) [inline container/appmaster/cli/dcos-flink/main.go]: addressed

File: container/appmaster/cli/dcos-flink/main.go (hunk 5 above)
Comment (OhRobin) [inline container/appmaster/cli/dcos-flink/main.go]: addressed

File: container/appmaster/cli/dcos-flink/main.go (hunk 1 above)
Comment (OhRobin) [inline container/appmaster/cli/dcos-flink/main.go]: removed upload. will not be part of this PR

File: container/appmaster/cli/dcos-flink/main.go (hunk 4 above)
Comment (OhRobin) [inline container/appmaster/cli/dcos-flink/main.go]: https://github.com/dcos/examples/pull/199/files

**Comment by OhRobin** [comment]
https://github.com/dcos/examples/pull/199/files

File: container/appmaster/cli/dcos-flink/main.go [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+	"gopkg.in/alecthomas/kingpin.v2"
+	"fmt"
+	"log"
+)
+
+
+func main() {
+	app := cli.New()
+
+	handleListSection(app)
```
Comment (joerg84) [inline container/appmaster/cli/dcos-flink/main.go]: we could make the handler names more descriptive (list what...), but I will change that.

File: container/appmaster/cli/dcos-flink/main.go [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+	handleJobSection(app)
+	handleRunSection(app)
+	handleCancelSection(app)
+  handleJarsSection(app)
```
Comment (joerg84) [inline container/appmaster/cli/dcos-flink/main.go]: indent

Comment (OhRobin) [inline container/appmaster/cli/dcos-flink/main.go]: this is odd because it shows up correctly in my text editor
